apps/worker/src/app/workflow/usecases/run-job/run-job.usecase.ts

```diff
@@ -50,13 +50,18 @@ export class RunJob {
       Logger.error(e, 'RunJob', LOG_CONTEXT);
     }
 
-    const canceled = await this.delayedEventIsCanceled(job);
-    if (canceled) {
+    const { canceled, activeDigestFollower } = await this.delayedEventIsCanceled(job);
+
+    if (canceled && !activeDigestFollower) {
       Logger.verbose({ canceled }, `Job ${job._id} that had been delayed has been cancelled`, LOG_CONTEXT);
 
       return;
     }
 
+    if (activeDigestFollower) {
+      job = this.assignNewDigestExecutor(activeDigestFollower);
+    }
```

Comment:
Good catch! We definitely should because the transaction/job ids are updated.
I will re-assign the logger if we re-assign the job. and move the new relic lower because once the custom attribute has been created and reported to new relic with a value, the value cannot be changed or updated.